src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php

```diff
@@ -167,4 +167,61 @@ protected function instantiateController($class)
     {
         return new $class();
     }
+
+    private function getControllerError($callable)
+    {
+        if (is_string($callable)) {
+            if (!function_exists($callable)) {
+                return sprintf('Function "%s" does not exist.', $callable);
+            }
+
+            if (false !== strpos($callable, '::')) {
+                $callable = explode('::', $callable);
+            }
+        }
+
+        if (!is_array($callable)) {
+            return sprintf('Invalid type for controller given, expected string or array, got "%s".', gettype($callable));
+        }
+
+        if (2 !== count($callable)) {
+            return sprintf('Invalid format for controller, expected array(controller, method) or controller::method.');
+        }
+
+        list($controller, $method) = $callable;
+
+        if (is_string($controller) && !class_exists($controller)) {
```

Comment:
It shouldn't, because the service id won't ever be passed to this function. When you make a type, the error will come from [ControllerResolver.php#L69](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php#L69) when the service doesn't exist